projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid-state.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class IgxPivotGridStateService {
```

Comment:
I don't think we need a whole new service on the root level. The pivot selector already has a reference to the related pivot and you can get the information you need from the config or modify it as needed directly. There's no need to go through a global service.

The service, as it is now, might even cause more issues, since it would be a single shared instance between grids on the page but it doesn't hold a reference to its owner. So if there are multiple grids with the same memberName it might return incorrect results (result added from another pivot on the page).